src/codec.cairo

```diff
@@ -102,24 +105,32 @@ pub impl EncodeWitness of Encode<Span<ByteArray>> {
 pub impl EncodeTx of Encode<Transaction> {
     fn encode_to(self: Transaction, ref dest: ByteArray) {
         self.version.encode_to(ref dest);
-        if (self.is_segwit) {
-            dest.append_byte(0); // marker
-            dest.append_byte(1); // flag
-        }
         self.inputs.encode_to(ref dest);
         self.outputs.encode_to(ref dest);
+        self.lock_time.encode_to(ref dest);
+    }
 
-        if (self.is_segwit) {
-            for txin in self.inputs {
-                txin.witness.encode_to(ref dest);
-            };
-        }
+    fn encode_to_wtx(self: Transaction, ref dest: ByteArray) {
```

Comment:
instead of adding another method, just handle include_witness argument inside `encode_to`